docs/src/main/asciidoc/mongo-guide.adoc

```diff
@@ -0,0 +1,599 @@
+
+`quarkus.mongodb.server-selection-timeout`::
+How long in milliseconds the driver will wait for server selection to succeed before throwing an exception.
+
+`quarkus.mongodb.local-threshold`::
+When choosing among multiple MongoDB servers to send a request, the driver will only send that request to a server whose ping time in milliseconds is less than or equal to the server with the fastest ping time plus the local threshold.
+
+`quarkus.mongodb.heartbeat-frequency`::
+The frequency in milliseconds that the driver will attempt to determine the current state of each server in the cluster.
+
+`quarkus.mongodb.read-preference`::
+Configures the read preferences. Supported values are: `primary|primaryPreferred|secondary|secondaryPreferred|nearest`
+
+`quarkus.mongodb.max-wait-queue-size`::
+Configures the maximum number of concurrent operations allowed to wait for a server to become available. All further operations will get an exception immediately.
+
+=== Write concern
+`quarkus.mongodb.write-concern.safe`::
+Configures the safety.
+If set to `true` (the default): the driver ensures that all writes are acknowledged by the MongoDB server, or else throws an exception.
+If set to `false`: the driver does not ensure that all writes are acknowledged by the MongoDB server.
+
+`quarkus.mongodb.write-concern.journal`::
+Configures the journal writing aspect.
+If set to `true` (the default): the driver waits for the server to group commit to the journal file on disk.
+If set to `false`: the driver does not wait for the server to group commit to the journal file on disk.
+
+`quarkus.mongodb.write-concern.w`::
+When set, the driver adds `w: wValue` to all write commands. It requires `safe` to be `true`.
+The value is typically a number, but can also be the `majority` string.
+
+`quarkus.mongodb.write-concern.retry-writes`::
+If set to `true`, the driver will retry supported write operations if they fail due to a network error. The default is `false`.
+
+`quarkus.mongodb.write-concern.w-timeout`::
+When set, the driver adds `wtimeout : ms`  to all write commands. It requires `safe` to be `true`. Time given in ms.
+
+=== Credentials
+`quarkus.mongodb.credentials.username`::
+Configures the username.
+
+`quarkus.mongodb.credentials.password`::
+Configures the password.
+
+`quarkus.mongodb.credentials.auth-mechanism`::
+Configures the authentication mechanism to use if a credential was supplied.
+The default is unspecified, in which case the client will pick the most secure mechanism available based on the
+sever version. For the GSSAPI and MONGODB-X509 mechanisms, no password is accepted, only the username.
+Supported values: `MONGO-CR|GSSAPI|PLAIN|MONGODB-X509`
+
+`quarkus.mongodb.credentials.auth-source`::
+Configures the source of the authentication credentials.
+This is typically the database that the credentials have been created. The value defaults to the database
```

Comment:
`the database that the credentials have been created` hmmm, `that` is weird here.

Reply:
Updated. This was actually in the upstream code...